keyboards/marksard/leftover30/keymaps/default/keymap.c

```diff
@@ -0,0 +1,214 @@
+  ),
+
+  [_ADJUST] = LAYOUT_all(
+  //,-----------------------------------------------------------------------------------------------------------.
+                 RESET,  RGBRST, AG_NORM, AG_SWAP, XXXXXXX, KC_HOME, KC_PGDN, KC_PGUP,  KC_END,  KC_INS, KC_PSCR,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+               RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, XXXXXXX, KC_MS_L, KC_MS_D, KC_MS_U, KC_MS_R,     KC_NLCK,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+               RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, XXXXXXX, KC_BTN1, KC_BTN2, XXXXXXX, XXXXXXX,     KC_CAPS,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      _______, _______,                                _______,                                 _______, KC_CAPS
+  //`-----------------------------------------------------------------------------------------------------------'
+  )
+};
+
+uint16_t get_tapping_term(uint16_t keycode) {
+  switch (keycode) {
+    case KC_SPRA:
+      return TAPPING_LAYER_TERM;
+    default:
+      return TAPPING_TERM;
+  }
+}
+
+int RGB_current_mode;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+  bool result = false;
+  switch (keycode) {
+    case KANJI:
+      if (record->event.pressed) {
+        if (keymap_config.swap_lalt_lgui == false) {
+          register_code(KC_LANG2);
+        } else {
+          SEND_STRING(SS_LALT("`"));
+        }
+      } else {
+        unregister_code(KC_LANG2);
+      }
+      break;
+#ifdef RGBLIGHT_ENABLE
+    //led operations - RGB mode change now updates the RGB_current_mode to allow the right RGB mode to be set after reactive keys are released
+    case RGB_MOD:
+        if (record->event.pressed) {
+        rgblight_mode(RGB_current_mode);
+        rgblight_step();
+        RGB_current_mode = rgblight_config.mode;
+        }
+    break;
+    case RGBRST:
+        if (record->event.pressed) {
+        eeconfig_update_rgblight_default();
+        rgblight_enable();
+        RGB_current_mode = rgblight_config.mode;
+        }
+    break;
+#endif
+    default:
+      result = true;
+      break;
+  }
+
+  return result;
+}
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (IS_LAYER_ON(_ADJUST)) {
+          if (clockwise) {
+              rgblight_increase_hue_noeeprom();
+          }
+          else
+          {
+              rgblight_decrease_hue_noeeprom();
+          }
+        }
+        else if (IS_LAYER_ON(_LOWER)) {
+          if (clockwise) {
+            SEND_STRING(SS_LCTL("y"));
+          } else {
+            SEND_STRING(SS_LCTL("z"));
+          }
+        }
+        else if (IS_LAYER_ON(_RAISE)) {
+          if (clockwise) {
+            SEND_STRING(SS_DOWN(X_LSFT)SS_TAP(X_DOWN)SS_UP(X_LSFT));
```

Comment:
This has a smaller compile size. 
```suggestion
            tap_code16(S(KC_DOWN));
```